### ISSUE TITLE
Intellisense unresolved function or method

### ISSUE THREAD

**Issue by tarnos12** [photonstorm/phaser#3529]
* A problem with my own code
* Phaser version 3.3.0

Hello,  

I have a small issue with intellisense.  
It works fine for most things so far, but I found one that doesn't work and that is:  
`this.add.particles('red');  `
particles is not found by my editor(it works tho), I was looking into the phaser code and found out that those methods are created with a function(that's probably it):  

```
GameObjectCreator.register('particles', function (config)
{
    var key = GetAdvancedValue(config, 'key', null);
    var frame = GetAdvancedValue(config, 'frame', null);
    var emitters = GetFastValue(config, 'emitters', null);

    //  frame is optional and can contain the emitters array or object if skipped
    var manager = new ParticleEmitterManager(this.scene, key, frame, emitters);

    var add = GetFastValue(config, 'add', false);

    if (add)
    {
        this.displayList.add(manager);
    }

    this.updateList.add(manager);

    return manager;
});
```  

I may be wrong and it's not very important, but could be useful if I managed to get a working intellisense for those methods otherwise my editor shows me unresolved function or method error as you can see in an image below.  
[Image](https://i.gyazo.com/9a3f62f7b18daaa6d25037719436aa02.png)

**Comment by photonstorm** [comment]
All Game Objects use the same method of registering themselves with the GameObjectCreator / Factory, so it's not the code above causing it. I would have thought intellisense comes from parsing the jsdocs rather than the code - might be worth trying against the `master` branch, which is version 3.4, to see if it still happens, as we've update a lot of docs.

**Comment by tarnos12** [comment]
Can you tell me exactly what to do? I am confused.  
I looked up github file:  
https://github.com/photonstorm/phaser/blob/master/dist/phaser.js  

And it says version 3.3.0 inside a file.  

```
/**
 * Creates a new Particle Emitter Manager Game Object and returns it.
 *
 * Note: This method will only be available if the Particles Game Object has been built into Phaser.
 *
 * @method Phaser.GameObjects.GameObjectCreator#particles
 * @since 3.0.0
 *
 * @param {object} config - [description]
 *
 * @return {Phaser.GameObjects.Particles.ParticleEmitterManager} The Game Object that was created.
 */
```  
Code above is the same for current version on github and the version I am using.(3.3.0)  
Could it be that my IDE is doing something wrong, but it can read other properties just fine, so far only particles is not working(probably more, but I am basing it on the first example from the website)

**Comment by photonstorm** [comment]
You need to build it.

**Comment by tarnos12** [comment]
I have no idea what it means to build it.  
Can you be more precise?  
Do you want me to install phaser with npm maybe?

**Comment by ampled** [comment]
If you want to be able to test the latest branch I recommend using a package manager like npm or yarn (my preference) with your project. Here's an example project that does that: https://github.com/photonstorm/phaser3-project-template

You would have to install the latest master commit with "yarn add https://github.com/photonstorm/phaser.git#master"

if the latest commit does not have the "built" version included my guess is you would have to navigate to \<your project\>/node_modules/phaser with your command line and run "yarn install" and "yarn build" and that should be it.

**Comment by ampled** [comment]
@tarnos12 version 3.4 is out now, so you can see if intellisense has improved or not

**Comment by tarnos12** [comment]
I think it's my IDE that is not working properly. Using cdn and downloading library via IDE is not fixing it.  
So, I decided to download phaser.js directly from a website, still same thing, but this time it didn't detect Phaser object as well, so my IDE told me to import it and I did, but particles method still is unresolved(It works, just IDE can't give me intellisense for it).  

I will try to install it using npm and see what happens.
